util/fusermount.c

```diff
@@ -1225,10 +1259,21 @@ static int should_auto_unmount(const char *mnt, const char *type)
 		goto out;
 
 	fd = open(mnt, O_RDONLY);
+
 	if (fd != -1) {
 		close(fd);
 	} else {
-		result = errno == ENOTCONN;
+		switch(errno) {
+		case ENOTCONN:
+			result = 1;
+			break;
+		case EACCES:
+			result = mnt_is_ENOTCONN_for_owner(mnt);
```

Comment:
I find the function name a bit confusing and only understood it, once I read the code. Maybe
something `?recheck_ENOTCONN_as_owner`

Reply:
Done!